CMakeLists.txt

```
@@ -1053,6 +1053,16 @@ if(HAVE_SIZEOF_LONG_LONG)
   set(HAVE_LL 1)
 endif()
 
+# Check for atomics support
+check_c_source_compiles(
+  "
+  int main(int, char**)
+  {
+    _Atomic int i = 0;
+    return 0;
```

```suggestion
  #include <stdatomic.h>
  
  int main(void)
  {
    atomic_int i = 0;
    return atomic_load(&i);
```

I would personally change the test to this, so there is no chance of warnings being raised here. This is enough to not make Clang emit anything with `-Weverything`.

---

Sure. I'll change the check in the Autotools code as well then.

---

is `atomic_int` really C11? I can only find references to it in C++ docs?